From 8e51bd177c90113a10d66726bb621fd4d33f7880 Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
Date: Fri, 7 Aug 2020 10:20:56 +0700
Subject: [PATCH] arm64: dts: r8a774b1-hihope-rzg2n(-rev2): Re-order CMA area
 and remove lossy when enable ECC Full Single channel

Lossy can't work with ECC, so we remove the lossy area.
Also move up CMA and MMP area to fix the memory block for data area.

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 .../dts/renesas/r8a774b1-hihope-rzg2n-rev2.dts     | 14 ++++----------
 .../boot/dts/renesas/r8a774b1-hihope-rzg2n.dts     | 14 ++++----------
 2 files changed, 8 insertions(+), 20 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-rev2.dts b/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-rev2.dts
index 9bdc7636c460..5d5076099b5c 100644
--- a/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-rev2.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-rev2.dts
@@ -29,17 +29,11 @@
 		#size-cells = <2>;
 		ranges;
 
-		/* device specific region for Lossy Decompression */
-		lossy_decompress: linux,lossy_decompress@54000000 {
-			no-map;
-			reg = <0x00000000 0x54000000 0x0 0x03000000>;
-		};
-
 		/* global autoconfigured region for contiguous allocations */
-		global_cma: linux,cma@58000000 {
+		global_cma: linux,cma@50000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x58000000 0x0 0x20000000>;
+			reg = <0x00000000 0x50000000 0x0 0x20000000>;
 			linux,cma-default;
 		};
 
@@ -47,13 +41,13 @@
 		mmp_reserved: linux,multimedia@70000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x78000000 0x0 0x10000000>;
+			reg = <0x00000000 0x70000000 0x0 0x10000000>;
 		};
 	};
 
 	mmngr {
 		compatible = "renesas,mmngr";
-		memory-region = <&mmp_reserved>, <&lossy_decompress>;
+		memory-region = <&mmp_reserved>;
 	};
 
 	mmngrbuf {
diff --git a/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n.dts b/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n.dts
index ca7ca362efa4..e99ceba89550 100644
--- a/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n.dts
@@ -29,17 +29,11 @@
 		#size-cells = <2>;
 		ranges;
 
-		/* device specific region for Lossy Decompression */
-		lossy_decompress: linux,lossy_decompress@54000000 {
-			no-map;
-			reg = <0x00000000 0x54000000 0x0 0x03000000>;
-		};
-
 		/* global autoconfigured region for contiguous allocations */
-		global_cma: linux,cma@58000000 {
+		global_cma: linux,cma@50000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x58000000 0x0 0x20000000>;
+			reg = <0x00000000 0x50000000 0x0 0x20000000>;
 			linux,cma-default;
 		};
 
@@ -47,13 +41,13 @@
 		mmp_reserved: linux,multimedia@70000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x78000000 0x0 0x10000000>;
+			reg = <0x00000000 0x70000000 0x0 0x10000000>;
 		};
 	};
 
 	mmngr {
 		compatible = "renesas,mmngr";
-		memory-region = <&mmp_reserved>, <&lossy_decompress>;
+		memory-region = <&mmp_reserved>;
 	};
 
 	mmngrbuf {
-- 
2.17.1

